### ISSUE TITLE
Argo CD stuck indefinitely while refreshing an App

### ISSUE THREAD

**Issue by gravops** [argoproj/argo-cd#15519]
Checklist:

* [ Y] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ Y] I've included steps to reproduce the bug.
* [ Y] I've pasted the output of `argocd version`.

**Describe the bug**

We are getting issues on multiple EKS clusters where Apps get stuck in refreshing state indefinitely.

**To Reproduce**
1. Enable Auto Sync.
2. Deploy a new App.

**Expected behavior**
App will get stuck in refreshing state indefinitely.
Even when we try to use command argocd app terminate-op APP-NAME, it doesn't help.
**Screenshots**

**Version**
v2.5.1

**Logs**

In logs we found that it times out after 3 mins, then again it retries to refresh the app status and again fails. This keeps on happening in loop infinitely.:
time="2023-09-15T08:04:19Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:04:04Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:03:19Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:03:04Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:04:34Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:05:05Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:05:20Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:05:35Z" level=info msg="Refreshing app status (controller refresh requested), level (1)" application=argocd/XXXX
time="2023-09-15T08:06:05Z" level=info msg="Refreshing app status (comparison expired, requesting refresh. reconciledAt: 2023-09-15 08:03:04 +0000 UTC, expiry: 3m0s), level (2)" application=argocd/XXXX

**Comment by jannfis** [comment]
Can you reproduce this issue with an actual (i.e. supported) version of Argo CD?

Please note that v2.5.1 is out of support for a while.

**Comment by shinebayar-g** [comment]
I was able to get rid of the stuck Refresh by deleting the Application temporarily. (Make sure you select **Non-cascading** option, or else you'll delete your k8s resources!) Restarting argocd containers didn't help.

**Comment by nicraMarcin** [comment]
I have the same issue with v2.7.3

**Comment by cultpony** [comment]
I've experienced the same issue with some apps I deployed, mainly Helm apps. Straight deploys from a rendered git repository seem to have no issues, and most Helm apps are fine. Specifically the signoz Helm charts repeatedly experience that problem. Using a non-cascading delete and recreation fixes the issue until the next time refresh is clicked manually.

**Comment by shinebayar-g** [comment]
> mainly Helm apps.

Yep. I also noticed this behavior happens usually for Helm charts. Like ingress-nginx & kube-prometheus-stack is always stuck at refreshing😕

**Comment by pimenas** [comment]
Happens to me also with the following helm charts: rabbitmq, haproxy and mongodb

However, if I put these on one Application with multiple sources, they seem to work fine, so maybe it's a workaround for a few people.

**Comment by acelinkio** [comment]
Also experiencing this issue where applications get stuck `Refreshing` on ArgoCD v2.9.3.  

Seeing this with ~40 applications.  Some of those are App of Apps managing other applications.  I am able to recreate the issue pretty consistently just by going into the UI and clicking `Refresh Apps` -> `All` -> `Refresh`.  Maybe there is an issue with the parent App of Apps and child App refreshing at the same time?

This is particularly cumbersome because there is no way to refresh the application as its hung in an infinite `Refreshing` state.  One way around is to go to the Application custom resource and delete the status field.  Triggering hard refresh to rebuild the state.

**Comment by FredrikAugust** [comment]
Are you using `valuesObject`? That often causes applications to get stuck refreshing indefinitely for me due to https://github.com/argoproj/argo-cd/issues/15126

**Comment by pimenas** [comment]
> Are you using `valuesObject`? That often causes applications to get stuck refreshing indefinitely for me due to #15126

Yes I do

**Comment by FredrikAugust** [comment]
@pimenas Could you try to convert your 

```
valuesObject:
  x: 1
  y: b
```

```
values: |
  x: 1
  y: b
```

And see if that resolves the issue?

You might have to restart ArgoCD, but that I'm not sure of.

**Comment by pimenas** [comment]
> @pimenas Could you try to convert your
> 
> ```
> valuesObject:
>   x: 1
>   y: b
> ```
> 
> ```
> values: |
>   x: 1
>   y: b
> ```
> 
> And see if that resolves the issue?
> 
> You might have to restart ArgoCD, but that I'm not sure of.
> 
> If that fixes it, I think this is a duplicate of #15126.

Yes, just tried it, and changing to 'values: |' fixes it, so it should be a duplicate of #15126 as you mention

**Comment by raweber42** [comment]
I was experiencing the same issue.

I had 13 small replicas of the application controller and reduced it to 6 with double the resources in order to reduce overhead and improve sharding. After applying the new configuration, all application controllers came back up, but when I went to the UI and refreshed an App, it got stuck in `Refreshing` state.

I restarted everything: application controller, repo server and server, nothing helped.

But than the strangest thing happened: I checked the corresponding apps with the `argocd` cli and I didn't see any refresh or sync going on (basically contradicting what I saw in the UI).

So I simply logged out and back into argocd and *_magically_* 🪄 the problem disappeared 😅 

Just wanted to share this, it might help someone.
